qiskit/compiler/assembler.py

```diff
@@ -102,7 +102,7 @@ def assemble(
         memory: If ``True``, per-shot measurement bitstrings are returned as well
             (provided the backend supports it). For OpenPulse jobs, only
             measurement level 2 supports this option.
-        max_credits: Maximum credits to spend on job. Default: 10
+        max_credits: DEPRECATED Maximum credits to spend on job. Default: 10
```

Comment:
In this function, we also need to make sure that we issue a deprecation warning in the function itself.  The test you added of this passes because further down the call tree, `RunConfig` issues a deprecation warning, but user code won't show the warning by default, because Python considers that a deprecated call in the library.  Instead, `assemble` should issue its own warning if `max_credits is not None`, and then set `max_credits = None`, to avoid triggering the warning a second time.